### PR TITLE
Add io.github.face_hh.Napture

### PR DESCRIPTION
**Pull request by QuinceTart10**
### Please confirm your submission meets all the criteria

- [x] Please describe your application briefly.
  - Napture is a simple browser using a modified version of the http protocol called buss, it uses Lua instead of JavaScript and has support for most HTML and CSS tags, it uses a custom DNS found at api.buss.lol.
- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I have [built][build] and tested the submission locally.
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** Screenshot of me asking the upstream developer, Face, whether I publish his app on Flathub and him responding with "Yes"
- [x] The domain used for the application ID is controlled by the application developers either directly or through the code hosting (e.g. GitHub, GitLab, SourceForge, etc.). The [application id guidelines][app-id] are followed.
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission
[app-id]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id

### CODE REVIEW

**Comment by creeperita09** [comment]
I agree this

File: io.github.face_hh.Napture.json [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        "org.freedesktop.Sdk.Extension.rust-stable"
+    ],
+    "command": "napture",
+    "finish-args": [
+        "--share=ipc",
+        "--socket=fallback-x11",
+        "--socket=wayland",
+        "--device=dri",
+        "--share=network",
+        "--own-name=org.bussin.napture"
```
Comment (bbhtt) [inline io.github.face_hh.Napture.json:16]: Why is the busname different than the appid? If the developer does not control the bussin.org then the bus name should be changed to match the appid.

If they do, the id should be `org.bussin.napture`.

File: data/io.github.face_hh.Napture.desktop
Comment (bbhtt) [inline data/io.github.face_hh.Napture.desktop:1]: This should be managed upstream

File: io.github.face_hh.Napture.json [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            "name": "lua",
+            "buildsystem": "simple",
+            "build-commands": [
+                "make all install",
+                "install -Dm644 lua.pc /app/lib/pkgconfig/lua.pc"
+            ],
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "https://www.lua.org/ftp/lua-5.4.6.tar.gz",
```
Comment (bbhtt) [inline io.github.face_hh.Napture.json:35]: This should use shared-modules https://github.com/flathub/shared-modules/blob/master/lua5.4/lua-5.4.json

If you need 5.4.6 I can update it there.

File: io.github.face_hh.Napture.json [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    ],
+    "build-options": {
+        "append-path" : "/usr/lib/sdk/rust-stable/bin",
+        "env" : {
+            "CARGO_HOME" : "/run/build/napture/napture/cargo"
+        }
+    },
+    "modules": [
+        {
+            "name": "lua",
```
Comment (hfiguiere) [inline io.github.face_hh.Napture.json:26]: Why not using the shared-modules https://github.com/flathub/shared-modules/tree/master/lua5.4

File: io.github.face_hh.Napture.json [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+            ],
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/face-hh/webx.git",
+                    "tag": "v1.0.1",
+                    "commit": "65e360016d608f6440ab28bd2bf9577ea623d25e"
+                },
+                {
+                    "type": "dir",
```
Comment (hfiguiere) [inline io.github.face_hh.Napture.json:71]: don't use `type: dir`

File: io.github.face_hh.Napture.json (hunk 2 above)
Comment (QuinceTart10) [reply io.github.face_hh.Napture.json:35]: To be honest, I didn't even know it existed, I'll incorporate it into the manifest once I get to it.

File: io.github.face_hh.Napture.json [hunk 5]
```diff
@@ -0,0 +1,79 @@
+{
+    "app-id": "io.github.face_hh.Napture",
```
[the remark this answers is not in the document]
Comment (QuinceTart10) [reply io.github.face_hh.Napture.json:2]: Web X actually consists of 2 parts: the DNS system and the browser, Napture

**Comment by bbhtt** [comment]
Icon is missing 

image

**Comment by bbhtt** [comment]
Doesn't seem to be working with the pages in the readme


https://github.com/flathub/flathub/assets/62639087/73865dc1-aa99-4f0c-94a0-06c0ee460000

**Comment by bw8686** [comment]
bot, build io.github.face_hh.Napture

**Comment by bbhtt** [comment]
You have to give me a working domain to try in the app. `buss://ohio.rizz` and the other one is still down.

**Comment by QuinceTart10** [comment]
> You have to give me a working domain to try in the app. `buss://ohio.rizz` and the other one is still down.

Try `buss://dingle.it`, `buss://register.it` or `buss://chat.it`

**Comment by QuinceTart10** [comment]
> Also this one is outputting a bunch of raw html in the terminal that it shouldn't

Doesn't happen to me on v1.2.1

**Comment by QuinceTart10** [comment]
bot, build io.github.face_hh.Napture

**Comment by bbhtt** [comment]
I don't know whose website is `buss://chat.it` but the content in it that is visible does not serve as a good advertisement for the protocol or a client for it.

Also a simple search shows too many nonsense websites. I don't see how hosting this is seems useful.

modified_31674307

**Comment by QuinceTart10** [comment]
> I don't know whose website is `buss://chat.it` but the content in it that is visible does not serve as a good advertisement for the protocol or a client for it.
> 
> Also a simple search shows too many nonsense websites. I don't see how hosting this is seems useful.
> 
> modified_31674307

1. Woah, I didn't see stuff like that on the site last time I checked it. That content even violates Web X's rules, so I've informed the site creator and Face.
2. The search engine seems to be a really simple one to quickly see what sites are there
3. Hosting Napture or Dingle?

**Pull request closed without merging**